databricks_cli/tunnel/api.py

```diff
@@ -0,0 +1,236 @@
+# Databricks CLI
+# Copyright 2021 Databricks, Inc.
+#
+# Licensed under the Apache License, Version 2.0 (the "License"), except
+# that the use of services to which certain application programming
+# interfaces (each, an "API") connect requires that the user first obtain
+# a license for the use of the APIs from Databricks, Inc. ("Databricks"),
+# by creating an account at www.databricks.com and agreeing to either (a)
+# the Community Edition Terms of Service, (b) the Databricks Terms of
+# Service, or (c) another written agreement between Licensee and Databricks
+# for the use of the APIs.
+#
+# You may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#    http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+import os
+from pathlib import Path
+
+import requests
+import tornado
+from cryptography.hazmat.primitives.asymmetric import rsa
+from tornado.ioloop import IOLoop
+
+from databricks_cli.clusters.api import ClusterApi
+from databricks_cli.commands.api import CommandApi, ExecutionContextApi
+from databricks_cli.tunnel.server import TunnelConfig, StreamingServer
+
+
+def generate_key_pair():
```

Comment:
nit: Can we add a comment here that this helper function returns (private key, public key) in case we switch the order at some point :)